### PR TITLE
WIP: Adding documentation on internationalization via hreflang

### PR DESCRIPTION
**Pull request by tcmg**
I've created a documentation page for how to structure a website with canonical and AMP documents that are translated into multiple languages.

Two examples are presented:
- A site with a canonical document and an AMP document in both English and French.
- A site with a canonical document, a mobile alternate document and an AMP document in both English and French.

The documentation includes 2 high level diagram images highlighting each of these examples.

The second example is not implemented, only described in the documentation and in the diagram.

Additionally there are unimplemented documents to highlight Japanese localisations as well.

This PR is a WIP for review and I welcome feedback regarding the approach of the example and the documentation's explanation.

### CODE REVIEW

File: src/10_Introduction/Internationalization.html [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+a corresponding AMP document. As per the official AMP Project documentation for
+[discovery and distribution](https://www.ampproject.org/docs/get_started/create/prepare_for_discovery)
+the English AMP document must include a link tag to the canonical document. Similarly, the
+canonical document includes a corresponding link tag to the AMP document.
+
+Additionally, this diagram includes French translations of both the canonical document
+and the AMP document. Therefore the documents must include link tags with hreflang
+attributes to connect the various alternative language versions of the canonical and
+AMP documents. The following rules apply:
+* Every canonical document must have a link tag with rel=amphtml pointing to
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: Can you please surround code snippets such `rel=amphtml` and `rel=canonical` with ` `` `.

File: static/internationalization/basic/fr/index.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
+
+    <link rel="amphtml" href="/internationalization/basic/fr/index.amp.html" />
+
+    <link rel="alternate" hreflang="en" href="/internationalization/basic/" />
+    <link rel="alternate" hreflang="fr" href="/internationalization/basic/fr/" />
+    <link rel="alternate" hreflang="ja" href="/internationalization/basic/ja/" />
+  </head>
+  <body>
+    <h1>French Desktop Document</h1>
```
Comment (sebastianbenz) [inline static/internationalization/basic/fr/index.html]: Would it make sense to show the included links directly in the samples? e.g.:

```
<p>This page contains the following links:</p>
<pre>
<link rel="amphtml" href="/internationalization/basic/fr/index.amp.html" />
    <link rel="alternate" hreflang="en" href="/internationalization/basic/" />
    <link rel="alternate" hreflang="fr" href="/internationalization/basic/fr/" />
    <link rel="alternate" hreflang="ja" href="/internationalization/basic/ja/" />
</pre>

**Comment by sebastianbenz** [comment]
@tcmg let me know when I should start the review

**Comment by tcmg** [comment]
@sebastianbenz I think it's ready for review. Thanks!

File: src/10_Introduction/Internationalization.html [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+websites often embrace to provide maximum reach for their content or services.
+
+For static websites, a common site structure technique for addressing this is
+known as 'hreflang'. Hreflang is a part of the meta data of a website and is used
+to annotate a page such that crawlers can discover alternative translations of
+a particular article. Search engines such as Google recognise websites that use hreflang.
+
+The hreflang notation is used with the link tag to refer to another document:
+
+`<link rel="alternate" hreflang="fr" href="/path/to/french.html" />`
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: Can you turn this into a code block using: '``` '. Looks better.

File: src/10_Introduction/Internationalization.html [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+You can [read more about hreflang here.](https://support.google.com/webmasters/answer/189077?hl=en)
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: Suggestion: remove this sentence and add the link the first time you mention hreflang.

File: src/10_Introduction/Internationalization.html [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+</head>
+<body>
+<!--
+
+## Introduction
+
+How to support hreflang and internationalization with AMP.
+
+## About Internationalization
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: suggestion: remove this heading as the following text is still an intro.

File: src/10_Introduction/Internationalization.html [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+However, it's important to correctly setup the hreflang relationships between
+AMP documents correctly. Especially when one considers the existing meta data
+relationships between Canonical documents and AMP documents. Below are two examples
+of how websites might structure their hreflang relationships for Canonical and AMP
+documents across an article.
+
+## Example 1: Canonical + AMP
+
+The most common site structure pattern is to support translations of an article
+with a canonical document and a corresponding AMP document.
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: Suggestion: rephrase to to

"A common site structure pattern is pairing non-AMP canonical documents with AMP documents."

File: src/10_Introduction/Internationalization.html [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+The following diagram illustrates this structure at a high level:
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: s/this structure at a high level/how to provide translations for this structure

**Comment by jmadler** [comment]
Great work!  Looks like this PR resolves #127 and #122 

If you add a hebrew or arabic copy (with appropriate changes made), you could knock out #123 as well

**Comment by tcmg** [comment]
@sebastianbenz I've implemented all the proposed changes except moving the canonical/hreflang away from being relative.

I've also made the article a draft.

I tried using <%host%> in the sample file but it didn't get replaced in the gulp process. This seems to work for the doc pages, do the samples themselves not support it?

**Comment by tcmg** [comment]
@jmadler Challenge accepted!

Would you be happy with that sample being a separate example but still listed under the same Internationalization page as the current hreflang examples?

I can still two options:

#1 - All internationalization examples go into a single documentation page "Internationalization", i.e. the samples would be: basic hreflang, advanced hreflang (+mobile alternate) and RLT example.

Or #2 - We break Internationalization out into a new category (on the same parity as Introduction, Components, Advanced, Samples & Templates) and then each of these samples appear as separate documentation pages.

Thoughts @sebastianbenz as well?

**Comment by sebastianbenz** [comment]
I think the RTL one is not really a sample that needs to be documented, but is more a proof that it works (@jmadler correct me if I'm wrong). In that case I think simply providing another translation should be fine.

**Comment by tcmg** [comment]
@sebastianbenz I just wanted to check in. I think the last time we chatted about this you were going to look into ways of bringing the template variables like <%host%> into the static directory's HTML files?

I experimented with this in the gulpfile.js but I'm not an expert on gulp and it didn't seem to work. If you can give me some direction though I can try further on my own.

Thanks!

**Comment by tcmg** [comment]
@kul3r4 I just chatted with Sebastian. He is on vacation, would you be able to assist with this instead?

Basically, I want to find a way of accessing the template variables like <%host%> from the files that are processed/generated in the /static directory of the project. In addition to the /src directory.

**Comment by kul3r4** [comment]
No worries, will take a look.

**Comment by kul3r4** [comment]
@tcmg qq, why these files cannot live in the src folder?

**Comment by tcmg** [comment]
I don't have a solid answer sorry.

I think @sebastianbenz originally proposed I put them under static/ but if
they can be moved to src/ instead then that's good with me.

**Comment by tcmg** [comment]
Hey @sebastianbenz, can you please review?

Thanks!

File: src/10_Introduction/Internationalization.html [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+`hreflang` attribute for each alternative language version of the canonical document.
+* Every AMP document must have a link tag with `rel=alternate` and an
+`hreflang` attribute for each alternative language version of the AMP document.
+
+Therefore, given an article with X language versions of itself
+(including English) each canonical document would include:
+* 1 link tag with `rel=amphtml`
+* And X link tags with `rel=alternate` and the `hreflang` attribute
+
+[View the example here.](/internationalization/basic/)
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: suggestion: turn this into a button as it makes the link more obvious

```
<a href="/internationalization/basic/" class="ampstart-btn ampstart-btn-secondary caps text-decoration-none inline-block"> View the Example</a>
```

File: src/10_Introduction/Internationalization.html [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+The same rules from the previous example apply in this example with the addition of
+the following rules:
+* Every canonical document must have a link tag with `rel=alternate` and no `hreflang` attribute
+pointing to its matching alternate mobile document in the same language.
+* Every alternate mobile document must have a link tag with `rel=canonical` pointing to
+its matching canonical desktop document, again, in the same language.
+* Every alternate mobile document must have a link tag with `rel=alternate` and an
+`hreflang` attribute for each alternative language version of the alternate mobile document.
+
+[View the example here.](/internationalization/alternate/)
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: same as above, turn this into a button: 

```
<a href="/internationalization/alternate/" class="ampstart-btn ampstart-btn-secondary caps text-decoration-none inline-block"> View the Example</a>
```

File: src/10_Introduction/Internationalization.html [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+</head>
+<body>
+<!--
+
+## Introduction
+
+How to support [hreflang](https://support.google.com/webmasters/answer/189077?hl=en) and internationalization with AMP.
+
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: nit: remove empty line

File: src/10_Introduction/Internationalization.html [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+Internationalization and localization of content is an important feature that
+websites often embrace to provide maximum reach for their content or services.
+
+For static websites, a common site structure technique for addressing this is
+known as 'hreflang'. Hreflang is a part of the meta data of a website and is used
+to annotate a page such that crawlers can discover alternative translations of
+a particular article. Search engines such as Google recognise websites that use hreflang.
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: nit: remove empty line

File: src/10_Introduction/Internationalization.html [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+However, it's important to correctly setup the hreflang relationships between
+AMP documents correctly. Especially when one considers the existing meta data
+relationships between Canonical documents and AMP documents. Below are two examples
+of how websites might structure their hreflang relationships for Canonical and AMP
+documents across an article.
+
+## Example 1: Canonical + AMP
+
+A common site structure pattern is pairing non-AMP canonical documents with AMP documents.
+
```
Comment (sebastianbenz) [inline src/10_Introduction/Internationalization.html]: nit: remove empty line

**Pull request merged**